SIGSEGV: segmentation violation

When running latest Pomerium authenticate (and Authorize) containers (pomerium/pomerium:latest with SERVICES set to authenticate or authorize), I'm encountering the following error on both:

```
10:37PM INF cmd/pomerium version=v0.1.0+357b202
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xae55d4]

goroutine 1 [running]:
main.main()
	/go/src/github.com/pomerium/pomerium/cmd/pomerium/main.go:64 +0x374
```

```
{"level":"info","version":"v0.1.0+357b202","time":"2019-07-26T22:38:22Z","message":"cmd/pomerium"}
{"level":"debug","route":"test.<mydomain-i-removed-from-this>","domain":"<mydomain-i-removed-from-this>","time":"2019-07-26T22:38:22Z","message":"add domain"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xae55d4]

goroutine 1 [running]:
main.main()
	/go/src/github.com/pomerium/pomerium/cmd/pomerium/main.go:64 +0x374
```

In this case, both containers are running in a Kubernetes environment - and appear to have been working for the last few months, until maybe earlier this week when pulling a newer image.

Curiously - the proxy service (SERVICES=proxy) seems unaffected

**To Reproduce** Steps to reproduce the behavior:

Pull latest Pomerium image

**Expected behavior**

Authenticate and Authorize services do not SIGSEGV, and runs normally.

**Environment:**

- Pomerium version (retrieve with `pomerium --version`): "v0.1.0+357b202
- Server Operating System/Architecture/Cloud: Kubernetes on baremetal, Centos 7 hosts

Configuration file(s):

Deployment for authenticate:
```text
apiVersion: apps/v1
kind: Deployment
metadata:
  name: pomerium-authenticate
  labels:
    app: pomerium-authenticate
  namespace: pomerium
spec:
  replicas: 1
  selector:
    matchLabels:
      app: pomerium-authenticate
  template:
    metadata:
      labels:
        app: pomerium-authenticate
    spec:
      containers:
        - image: pomerium/pomerium:latest
          name: pomerium-authenticate
          imagePullPolicy: Always
          ports:
            - containerPort: 443
              name: https
              protocol: TCP
          volumeMounts:
            - name: certs
              mountPath: "/etc/pomerium-certs/"
          env:
            - name: POMERIUM_DEBUG
              value: "true"
            - name: SERVICES
              value: authenticate
            - name: AUTHENTICATE_SERVICE_URL
              value: https://authenticate.<snipped>
            - name: IDP_PROVIDER
              value: google
            - name: IDP_PROVIDER_URL
              value: https://accounts.google.com
            - name: IDP_CLIENT_ID
              value: <snipped>.apps.googleusercontent.com
            - name: SHARED_SECRET
              valueFrom:
                secretKeyRef:
                  name: shared-secret
                  key: shared-secret
            - name: COOKIE_SECRET
              valueFrom:
                secretKeyRef:
                  name: cookie-secret
                  key: cookie-secret
            - name: IDP_CLIENT_SECRET
              valueFrom:
                secretKeyRef:
                  name: idp-client-secret
                  key: idp-client-secret
            - name: IDP_SERVICE_ACCOUNT
              valueFrom:
                secretKeyRef:
                  name: idp-service-account
                  key: idp-service-account
            - name: CERTIFICATE_FILE
              value: /etc/pomerium-certs/tls.crt
            - name: CERTIFICATE_KEY_FILE
              value: /etc/pomerium-certs/tls.key
          readinessProbe:
            httpGet:
              path: /ping
              port: 443
              scheme: HTTPS
          livenessProbe:
            httpGet:
              path: /ping
              port: 443
              scheme: HTTPS
            initialDelaySeconds: 10
            timeoutSeconds: 1
      volumes:
        - name: certs
          secret:
            secretName: cert-tls
            defaultMode: 256

```

Deployment for authorize
```text
apiVersion: apps/v1
kind: Deployment
metadata:
  name: pomerium-authorize
  labels:
    app: pomerium-authorize
  namespace: pomerium
spec:
  replicas: 1
  selector:
    matchLabels:
      app: pomerium-authorize
  template:
    metadata:
      labels:
        app: pomerium-authorize
    spec:
      containers:
        - image: pomerium/pomerium:latest
          name: pomerium-authorize
          imagePullPolicy: Always
          ports:
            - containerPort: 443
              name: https
              protocol: TCP
          volumeMounts:
            - name: certs
              mountPath: "/etc/pomerium-certs/"
          env:
            - name: POLICY
              valueFrom:
                configMapKeyRef:
                  name: policy
                  key: policy
            - name: SERVICES
              value: authorize
            - name: SHARED_SECRET
              valueFrom:
                secretKeyRef:
                  name: shared-secret
                  key: shared-secret
            - name: CERTIFICATE_FILE
              value: /etc/pomerium-certs/tls.crt
            - name: CERTIFICATE_KEY_FILE
              value: /etc/pomerium-certs/tls.key
          readinessProbe:
            httpGet:
              path: /ping
              port: 443
              scheme: HTTPS
          livenessProbe:
            httpGet:
              path: /ping
              port: 443
              scheme: HTTPS
            initialDelaySeconds: 10
            timeoutSeconds: 1
      volumes:
        - name: certs
          secret:
            secretName: cert-tls
            defaultMode: 256

```

Policy looks fairly normal, but would be mostly redacted if posted.

--- a/cmd/pomerium/main.go
+++ b/cmd/pomerium/main.go
@@ -61,8 +61,10 @@ func main() {
 	if err != nil {
 		log.Fatal().Err(err).Msg("cmd/pomerium: proxy")
 	}
-	defer proxy.AuthenticateClient.Close()
-	defer proxy.AuthorizeClient.Close()
+	if proxy != nil {
+		defer proxy.AuthenticateClient.Close()
+		defer proxy.AuthorizeClient.Close()
+	}
 
 	go viper.WatchConfig()
 

